reframe/frontend/executors/policies.py

```diff
@@ -27,6 +27,28 @@ def _cleanup_all(tasks, *args, **kwargs):
     tasks[:] = [t for t in tasks if t.ref_count]
 
 
+def format_duration(duration):
```

Comment:
I would make this a method of the task and rename it `pipeline_timings(phases)`, where `phases` is the list of phases or pseudo-phases whose timings you want to format. The idea is that you can call this a second time and print all the timings with the verbose log level.